deps/ReactantExtra/tblgen/jl-generators.cc

```diff
@@ -13,188 +13,461 @@
+      "export",  "false",    "finally",    "for",    "function", "global",
+      "if",      "import",   "let",        "local",  "macro",    "module",
+      "public",  "quote",    "return",     "struct", "true",     "try",
+      "using",   "while"};
+  if (modulename.has_value()) {
+    reservedKeywords.push_back(modulename.value());
   }
+  if (std::find(reservedKeywords.begin(), reservedKeywords.end(), name) !=
+      reservedKeywords.end()) {
+    name = name + "_";
+  }
+  // replace all .'s with _'s
+  std::replace(name.begin(), name.end(), '.', '_');
+  std::replace(name.begin(), name.end(), '-', '_');
+  return name;
+}
 
-  std::string formatDescription(mlir::tblgen::Operator op)
-  {
-    std::string description;
-    description = op.getDescription().str();
-    size_t pos = 0;
-    while (description[pos] == '\n')
-      ++pos;
-    size_t leading_spaces = 0;
-    while (description[pos++] == ' ')
-      ++leading_spaces;
-    if (leading_spaces)
-    {
-      std::string leading_spaces_str;
-      for (size_t i = 0; i < leading_spaces; ++i)
-        leading_spaces_str += "[ ]";
-      description = std::regex_replace(description, std::regex("\n" + leading_spaces_str), "\n");
-    }
-    description = std::regex_replace(description, std::regex(R"(\\)"), R"(\\)");
-    description = std::regex_replace(description, std::regex("(['\"$])"), "\\$1");
-    description = std::regex_replace(description, std::regex("(^|\n)(Example|Syntax):"), "$1# $2");
+} // namespace
+
+extern bool disableModuleWrap;
+
+template <typename VType>
+std::optional<VType> get(llvm::StringMap<VType> m, std::string k) {
+  auto entry = m.find(k);
+  return (entry != m.end()) ? std::optional<VType>(entry->getValue())
+                            : std::nullopt;
+}
+
+std::string removeNamespace(std::string s) {
+  auto pos = s.rfind("::");
+  if (pos >= s.length())
+    return s;
+  return s.substr(pos + 2);
+}
 
-    // remove trailing whitespaces and newlines
-    while (std::isspace(description.back())) {
-      description.pop_back();
+auto attribs = std::string();
+
+llvm::StringMap<std::string> attributeCache;
+
+std::string emitEnum(llvm::Record def, std::string dialect) {
+  EnumAttr e(def.isSubClassOf("EnumAttrInfo") ? def
+                                              : *def.getValueAsDef("enum"));
+  auto tableGenName = def.getName().str();
+  if (auto cached = get(attributeCache, tableGenName))
+    return *cached;
+
+  auto base = e.getBaseAttrClass();
+  auto enumJuliaType_ = e.getEnumClassName().str();
+  auto enumJuliaType = sanitizeName(enumJuliaType_);
+  auto juliaEnum = "@enumx " + enumJuliaType + ' ';
+  auto juliaStorage = enumJuliaType + "Storage";
+  enumJuliaType += ".T";
+  auto mlirAttributeDef = "IR.Attribute(e::" + enumJuliaType + ") = ";
+  auto isSpecialized = e.genSpecializedAttr();
+  if (!isSpecialized) { // parse the attribute using the name
+    auto juliaNameArray = juliaStorage + " = [";
```

Comment:
Can we make these consts?

```suggestion
    auto juliaNameArray = "const " + juliaStorage + " = [";
```

Reply:
sure!

Reply:
There are some improvement I want to add to this file, but I will prefer to add them in a following PR.